examples/httpbin-demo/tests/integration/conftest.py

```diff
@@ -31,5 +31,14 @@ def juju(request: pytest.FixtureRequest):
 
 @pytest.fixture(scope="session")
 def charm():
-    subprocess.check_call(["charmcraft", "pack"])  # noqa
-    return next(pathlib.Path(".").glob("*.charm"))
+    if "CHARM_PATH" in os.environ:
+        charm_path = pathlib.Path(os.environ["CHARM_PATH"])
+        if not charm_path.exists():
+            raise FileNotFoundError(f"Charm does not exist: {charm_path}")
+        return charm_path
+    charm_paths = list(pathlib.Path(".").glob("*.charm"))
+    if not charm_paths:
+        raise FileNotFoundError("No .charm file in current directory")
+    if len(charm_paths) > 1:
+        raise ValueError("More than one .charm file in current directory")
```

Comment:
Nit: might be helpful to show what they are, something like:
```suggestion
        raise ValueError("More than one .charm file in current directory: " + ', '.join(str(p) for p in charm_paths))
```

Reply:
Good idea, thanks. I've gone with this (easier to read after ruffing):

```py
path_list = ", ".join(str(path) for path in charm_paths)
raise ValueError(f"More than one .charm file in current directory: {path_list}")
```